internal/provider/resource_access_model.go

```diff
@@ -0,0 +1,269 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/juju/terraform-provider-juju/internal/juju"
+)
+
+func resourceAccessModel() *schema.Resource {
+	return &schema.Resource{
+		// This description is used by the documentation generator and the language server.
+		Description: "A resource that represent a Juju Access Model.",
+
+		CreateContext: resourceAccessModelCreate,
+		ReadContext:   resourceAccessModelRead,
+		UpdateContext: resourceAccessModelUpdate,
+		DeleteContext: resourceAccessModelDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceAccessModelImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"model": {
+				Description: "The name of the model for access management",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"users": {
+				Description: "List of users to grant access to",
+				Type:        schema.TypeList,
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"access": {
```

Comment:
If only `admin`, `write`, and `admin` are available we must use the `ExactlyOneOf` field here.

Reply:
Thanks for the feedback! I see that `ExactlyOneOf` relates to attribute paths, meaning one of the defined attributes of the schema has to be set and that differs from what is expected here, where we would like to validate a value for a specific attribute (`access`). That would be more suitable for `ValidateFunc` or `ValidateDiagFunc` - what do you think? I like the idea to limit what is supported in the API early on, I hope I'm on the right track.

Reply:
You're right. I think `ValidateFunc` should be enough here.

Reply:
Thanks for the confirmation, added!